### PR TITLE
[Feature] : Improve animation in explore section

### PR DESCRIPTION
**Pull request by Himanshushakyaa**
## Description

-   [Ticket Link](#) (if any)
-   improve animation in explore service section

## UI screenshot or Video

-   

https://github.com/primathonacademy/april-batch-live-project/assets/131392846/b2c76408-c1ff-48a5-87b6-babf289b597c

---

### General:

-   [x] **PR Size :** Confirming PR size < 20 files or pre-approved if > 20 files.
-   [x] **Self Review :** Confirming I have self-reviewed my PR
-   [x] **Peers Review :** - [ ] @Dev 1
-   [x] Confirming I have added all sensitive information-related should be inside .env files

### Client:

-   [ ] **Console :** Confirming no console errors in the UI for this update.

-   [ ] **Import order :** Confirming correct import order.

```
# External Import
import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { useDispatch, useSelector, shallowEqual } from 'react-redux';

# Internal Import
import { } from '../../Api/ApiRoutes';
import { } from '../../constants/AppConstant';
```

-   [x] **Re-renders :** Confirming minimal re-renders for new React components.

-   [x] **Network calls :** Confirming efficient network/API calls, minimized as needed.

-   [x] **Util function comments :** Confirming comprehensive comments for utility functions.

-   [x] **Image optimization :** Confirming I have checked that all images are optimized. If not, please use https://tinyjpg.com/ or https://vecta.io/nano.

-   [ ] **Dead code :** Confirming I have removed all comments code which is of no use OR written why I have kept the commented code.

### Confirming different viewport testing of the entire flow:

-   [x] Tested on 280px viewport
-   [x] Tested on 320px viewport
-   [x] Tested on 375px viewport
-   [x] Tested on 425px viewport
-   [x] Tested on 768px viewport
-   [x] Tested on 1024px viewport
-   [x] Tested on 1440px viewport
-   [x] Tested on 2560px viewport

### End-to-end testing:

-   [x] **Happy flow :** Successfully tested end-to-end happy flow, all working as expected.
-   [ ] **Error flow :** Tested UI flow with error scenarios, gracefully handled errors on UI
-   [ ] **Common component changed :** Tested change in common component across views to prevent breaking elsewhere.

### CODE REVIEW

File: src/pages/homepage/ExploreService.jsx [hunk 1]
```diff
@@ -1,44 +1,55 @@
...
     useGSAP(() => {
-        if (useWindowResize) {
-            gsap.registerPlugin(ScrollTrigger);
-            const cleanupScrollHandler = scrollHandler();
-            return () => {
-                cleanupScrollHandler();
-            };
-        }
+        const container = containerRef.current;
+        const images = container.querySelectorAll('.image-container');
```
Comment (ram-primathon) [inline src/pages/homepage/ExploreService.jsx]: Not allowed find a react way of doing this

**Review by ram-primathon — CHANGES_REQUESTED** [review]

File: src/pages/homepage/ExploreService.jsx [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
+import { ourServiceImage1, ourServiceImage2, ourServiceImage3, rightArrowIcon } from '../../assets';
 
 const ExploreService = () => {
     const containerRef = useRef(null);
+
     useGSAP(() => {
         const container = containerRef.current;
-        const images = container.querySelectorAll('.image-container');
-        gsap.from(images, {
+        gsap.from(container.querySelectorAll('.image-container'), {
```
Comment (ram-primathon) [inline src/pages/homepage/ExploreService.jsx:11]: this is same as previous

**Review by ram-primathon — CHANGES_REQUESTED** [review]

File: src/pages/homepage/ExploreService.jsx (hunk 2 above)
Comment (Himanshushakyaa) [reply src/pages/homepage/ExploreService.jsx:11]: sir iye mene isliye use kiya hai kioki iye animation 3 jagah use krna tha image-container class use krke

Comment (Himanshushakyaa) [reply src/pages/homepage/ExploreService.jsx:11]: isliye iye maping krke use krna pda

Comment (Himanshushakyaa) [reply src/pages/homepage/ExploreService.jsx:11]: or koi another method se nhi ho rha sir isliye aisa kiya

**Pull request closed without merging**